src/bin/sound.rs

```diff
@@ -92,3 +74,22 @@ fn main() {
         terminal.draw();
     }
 }
+
+fn generate_sine_wave(duration_secs: f32, frequency: f32) -> Vec<u8> {
```

Comment:
It's still not ideal to have this kind of unbounded dynamic memory allocation

But I'm not familiar with the i2s driver, maybe it can be done in a more streaming way with a smaller, statically allocated buffer?